daemon/daemon_unix.go

```diff
@@ -375,6 +390,13 @@ func verifyPlatformContainerSettings(daemon *Daemon, hostConfig *containertypes.
 		warnings = append(warnings, "IPv4 forwarding is disabled. Networking will not work.")
 		logrus.Warnf("IPv4 forwarding is disabled. Networking will not work")
 	}
+	if hostConfig.CgroupParent != "" && daemon.usingSystemd() {
+		// CgroupParent for systemd cgroup should be named as "xxx.slice"
+		length := len(hostConfig.CgroupParent)
+		if length <= 6 || hostConfig.CgroupParent[length-6:] != ".slice" {
```

Comment:
This sounds like systemd's bug, not sure it's a good idea to limit docker users to avoid such issue.